allneeds.lua

```diff
@@ -0,0 +1,81 @@
+-- print sum of all needs
+--[====[
+
+allneeds
+========
+Show which needs are high for all dwarfs.
+Weight and Focus are divided by the number of dwarves that have that need.
+
+]====]
+local ENUM = {}
+
+ENUM[0]  = "Socialize"
```

Comment:
You should use `df.need_type` instead - you're building your own copy of it here. (Side note: all of the built-in enum types provided by DFHack can be indexed by both ID and name. You're indexing by ID here, which is common, i.e. `df.need_type[0] == "Socialize"`, but the reverse, `df.need_type.Socialize == 0`, also works.)

Reply:
Actually, I'm curious now - how did you come up with this list? I was able to find [need_type](https://github.com/DFHack/df-structures/blob/0.44.12-r2/df.units.xml#L467) almost immediately in the XML files by searching for `Socialize` and `DrinkAlcohol`, and it looks like the names all match, so clearly you got them from somewhere... but I'm not sure where that was, and if it wasn't making the type (`need_type`) clear.

Reply:
So procrastination let me stumble upon this abandoned side project. 

I believe I found that list, but since I do not really understand how lua works at all (my experience is with python and C++) I made it work. 

I think the issue was I didn't know how to print the name of the enum as a string. 

maybe i'll fix this up over the summer in case others find it useful

Reply:
No worries, I've been getting side-tracked too. Like I said, most of the things I had comments on were specific to the "DFHack way" of doing things - the script itself looks good. If I get around to this before you do, I can certainly help clean it up (I'm just focused on getting a stable release finalized at the moment).